File: two_pointers/11_container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int max = 0;
        int left = 0, right = height.length - 1;
        while(left < right){
            int area = Math.min(height[left],height[right]) * (right - left);
            max = Math.max(max, area);
            // Only moves the lower one, to get more area
            if(height[left] < height[right]){
                left++;
            }else{
                right--;
            }
        }
        return max;
    }
}

File: two_pointers/11containerWithMostWater.js
/**
 * @param {number[]} heights
 * @return {number}
 */
function containerWithMostWater(heights) {
    let maxArea = 0;
    let start = 0, end = heights.length - 1;
    while (start < end) {
        const width = end - start;
        const height = Math.min(heights[start], heights[end]);
        maxArea = Math.max(maxArea, width * height);
        if (heights[start] < heights[end]) {
            start++;
        }
        if (heights[start] > heights[end]) {
            end--;
        }
    }
    return maxArea;
}